### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ArvidSilverlock**
Requesting a code review, would like suggestions on improvements that could be made.

Almost a ground-up rewrite that implements:
- A pre-processing step on the data to make it easier to use.
- Separation of all of the different types of generation into separate files to improve the separation of concerns.
- A fairly bare-bones wiki, complete rewrite in comparison to [wiki](https://github.com/ArvidSilverlock/Pilot.lua-Luau-LSP/tree/wiki) branch. The wiki still utilises most of the docusaurus template images and such.
- Improvements to the `pilot-generate` command by inverting the most common options (i.e., `--nodocs` instead of having to specify `--docs` every time).
- A hopefully better general code-structure.

### CODE REVIEW

**Comment by someoneidoknow** [comment]
LGTM (did not read at all)

File: .gitignore
Comment (lrockreal) [inline .gitignore:1]: `.lune/local` feels like something that should be in a user's gitignore, not the repository's

File: wiki/README.md
Comment (lrockreal) [inline wiki/README.md:1]: Wiki still seems to be work in progress. We shouldn't push this to master until it's ready

File: README.md
Comment (lrockreal) [inline README.md:1]: Looks good, is there a motive behind the `pilot-setup` -> `pilot-generate` change?

File: .vscode/extensions.json
Comment (lrockreal) [inline .vscode/extensions.json:1]: StyLua should be included here if it's being used as a default formatter in the workspace settings

File: .vscode/settings.json
Comment (lrockreal) [inline .vscode/settings.json:1]: Some of these e.g. `editor.formatOnSave` should be decided by the user's config, not the workspace. Also consider moving `prettier.*` to [.prettierrc](https://prettier.io/docs/en/configuration.html)

File: wiki/package.json
Comment (lrockreal) [inline wiki/package.json:1]: Consider changing the package name, adding an author, etc.

File: wiki/static/img/logo.svg
Comment (lrockreal) [inline wiki/static/img/logo.svg:1]: Consider changing the logo to something WOS related

File: wiki/package-lock.json
Comment (lrockreal) [inline wiki/package-lock.json:1]: 24k lines lmao

**Comment by eyecubed** [comment]
looks like a pull request with +27,799 lines added and 1947 lines removed

**Review by lrockreal — COMMENTED** [review]
Review is incomplete. Submitting anyway as this has already been merged

File: wiki/README.md
Comment (ArvidSilverlock) [reply wiki/README.md:1]: people are asking for a wiki

File: README.md
Comment (ArvidSilverlock) [reply README.md:1]: it being one command is just where the infrastructure led me

File: wiki/static/img/logo.svg
Comment (ArvidSilverlock) [reply wiki/static/img/logo.svg:1]: @someoneidoknow is handling that

File: .vscode/extensions.json
Comment (ArvidSilverlock) [reply .vscode/extensions.json:1]: How do you mean? It does include `JohnnyMorganz.stylua` already

Comment (lrockreal) [reply .vscode/extensions.json:1]: Oh wow I don't know how I missed that

**Pull request merged**